great_expectations/data_context/data_context/serializable_data_context.py

```diff
@@ -0,0 +1,447 @@
+from __future__ import annotations
+
+import logging
+import os
+import shutil
+import warnings
+from typing import Optional, Union
+
+from ruamel.yaml import YAML
+
+import great_expectations.exceptions as gx_exceptions
+from great_expectations.data_context.data_context.abstract_data_context import (
+    AbstractDataContext,
+)
+from great_expectations.data_context.templates import (
+    CONFIG_VARIABLES_TEMPLATE,
+    PROJECT_TEMPLATE_USAGE_STATISTICS_DISABLED,
+    PROJECT_TEMPLATE_USAGE_STATISTICS_ENABLED,
+)
+from great_expectations.data_context.types.base import (
+    CURRENT_GX_CONFIG_VERSION,
+    MINIMUM_SUPPORTED_CONFIG_VERSION,
+    AnonymizedUsageStatisticsConfig,
+    DataContextConfig,
+    DataContextConfigDefaults,
+)
+from great_expectations.data_context.util import file_relative_path
+
+logger = logging.getLogger(__name__)
+yaml = YAML()
+yaml.indent(mapping=2, sequence=4, offset=2)
+yaml.default_flow_style = False
+
+
+class SerializableDataContext(AbstractDataContext):
+    """
+    TODO - write docstring!
+    """
+
+    UNCOMMITTED_DIRECTORIES = ["data_docs", "validations"]
+    GX_UNCOMMITTED_DIR = "uncommitted"
+    BASE_DIRECTORIES = [
+        DataContextConfigDefaults.CHECKPOINTS_BASE_DIRECTORY.value,
+        DataContextConfigDefaults.EXPECTATIONS_BASE_DIRECTORY.value,
+        DataContextConfigDefaults.PLUGINS_BASE_DIRECTORY.value,
+        DataContextConfigDefaults.PROFILERS_BASE_DIRECTORY.value,
+        GX_UNCOMMITTED_DIR,
+    ]
+    GX_DIR = "great_expectations"
+    GX_YML = "great_expectations.yml"
+    GX_EDIT_NOTEBOOK_DIR = GX_UNCOMMITTED_DIR
+    DOLLAR_SIGN_ESCAPE_STRING = r"\$"
+
+    def __init__(
+        self, context_root_dir: str, runtime_environment: Optional[dict] = None
+    ) -> None:
+        self._context_root_directory = context_root_dir
+        super().__init__(runtime_environment=runtime_environment)
+
+    def _init_datasource_store(self):
+        pass
+
+    def _init_variables(self):
+        pass
+
+    def _save_project_config(self) -> None:
+        """
+        See parent 'AbstractDataContext._save_project_config()` for more information.
+
+        Explicitly override base class implementation to retain legacy behavior.
+        """
+        logger.debug("Starting DataContext._save_project_config")
```

Comment:
Is there any helpful info you can provide about the DataContext here? I'm fine with keeping it, but I'm wondering how useful this debug message is.

Reply:
done!